python/rmm/tests/test_rmm.py

```diff
@@ -216,6 +216,26 @@ def test_rmm_device_buffer_copy_from_host(hb):
     np.testing.assert_equal(expected, result)
 
 
+def test_rmm_device_buffer_copy_from_device():
```

Comment:
I originally had the `cuda_ary` objects as parameters below but it was causing a segfault and I wasn't quite sure why, here was the backtrace:
```
(cuda-gdb) bt
#0  0x000055555a9880b0 in ?? ()
#1  0x00007ffff04a5ea0 in rmm::mr::device_memory_resource::deallocate (stream=<optimized out>,
    bytes=<optimized out>, p=<optimized out>, this=<optimized out>)
    at ../include/rmm/mr/device/device_memory_resource.hpp:100
#2  rmm::device_buffer::~device_buffer (this=0x55555b4dae10, __in_chrg=<optimized out>)
    at ../include/rmm/device_buffer.hpp:251
#3  std::default_delete<rmm::device_buffer>::operator() (this=<optimized out>, __ptr=0x55555b4dae10)
    at /usr/include/c++/7/bits/unique_ptr.h:78
#4  std::unique_ptr<rmm::device_buffer, std::default_delete<rmm::device_buffer> >::~unique_ptr (
    this=0x7ffff0945fa8, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/unique_ptr.h:268
#5  __Pyx_call_destructor<std::unique_ptr<rmm::device_buffer, std::default_delete<rmm::device_buffer> > > (x=...) at rmm/_lib/device_buffer.cpp:334
#6  __pyx_tp_dealloc_3rmm_4_lib_13device_buffer_DeviceBuffer (o=0x7ffff0945f90)
    at rmm/_lib/device_buffer.cpp:20509
#7  0x0000555555647708 in dict_dealloc ()
    at /home/conda/feedstock_root/build_artifacts/python_1573054930886/work/Objects/dictobject.c:2017
```

Reply:
Sorry missed this when reviewing. Would it be possible to just use host array to parametrize and then convert those to device buffers within the test?

Reply:
Yes it would. I think the issue is that the parameter objects are created at some point during setup and then we have tests that finalize / reinitialize RMM and they break those objects. Will change to lambdas which should resolve this.

Reply:
Yeah that's what I was wondering too.

We could also just include things like `b"abc"` in `pytest.parametrize` and save the `to_device` call for the test. Either way. 🙂

Reply:
So there's two different objects being constructed, one is a numba device array and the other is an rmm.DeviceBuffer, and I wanted to keep both objects being tested. Just pushed a fix that uses a lambda which allows us to add additional objs less painfully.